planet/src/main/java/com/google/planet/servlets/EventServlet.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.planet.servlets;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.planet.data.Event;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.*; 
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles the game two truths and one lie */
+
+@WebServlet("/update-event")
+public class EventServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Wait 50 ms for datastore to update changes
+        try{
+            Thread.sleep(50);
```

Comment:
It feels counterintuitive to me that we are waiting for the datastore to finish updating at the beginning of a doFetch(). Is this a common practice? Can you move the sleep() to the end of doPost() (i.e. right before the redirect)?

Reply:
+1 this logic should be in the client, or the doPost should not return success until the datastore update is confirmed complete.

Reply:
That was my first approach but unfortunately it wasn't working when I tested it, will try it again. We are also switching to firebase instead of datastore later.

Reply:
I changed it back to doPost, it works most of the time so that's good